force-app/main/default/aura/CandidateGrade/CandidateGrade.cmp

```diff
@@ -4,6 +4,11 @@
     <aura:attribute name="displayGrade" type="Boolean" default="false" />
     <aura:handler name="init" value="{!this}" action="{!c.doInit}"/>
 
+    <ltng:require
```

Comment:
### Security (ISVs): CRUD/FLS violation
#### Severity
Critical

#### Finding
Method `createLogObjectRecords` can bypass CRUD \(on Create\) permission. Please make sure all proper checks are performed before accessing or updating data.

#### Why is this a problem?
CRUD and FLS are authorization settings that let Admins specify which objects and which fields on an object a user can access and modify. There are cases where developers use Visualforce or Lightning to display data derived from an SObject field in an indirect or processed form. In such scenarios CRUD and FLS should be manually enforced.

#### Help and documentation

* [Enforcing CRUD and FLS (Developer Documentation)](https://developer.salesforce.com/page/Enforcing\_CRUD\_and\_FLS) 
* [Improper Authorization (CWE)](https://cwe.mitre.org/data/definitions/285.html) 
* [Field-Level Security (Developer Documentation)](https://developer.salesforce.com/docs/atlas.en-us.securityImplGuide.meta/securityImplGuide/admin\_fls.htm) 
* [A5-Broken Access Control (OWASP)](https://owasp.org/www-project-top-ten/OWASP\_Top\_Ten\_2017/Top\_10-2017\_A5-Broken\_Access\_Control)